### ISSUE TITLE
My little suggestion

### ISSUE THREAD

**Issue by ShixiangWang** [SebKrantz/collapse#150]
Hi developer,

I am a user of dplyr and data.table, and it's very interesting to find an R package seems to have better performance than data.table. I saw there are many functions provided by this package and it's very hard and great work! After I check the function names, I have a little suggestion, maybe the whole package (not a part of it) should use a **very consistent way** to define function names. From the cheat sheet, I saw you mix `.`, snakecase and camel case in function names. In this case, I would use some of the functions (for performance) to get my work done but I would not like to spend my time to get familiar with the self-contained data processing ecosystem you built. I didn't say this package is not good, what I want to express is that this package has the potential to be great, similar to data.table. 

Best,
Shixiang

**Comment by SebKrantz** [comment]
Hi Shixiang, thanks for your suggestion. I have received comments like this before, and there are potentially small namespace changes I can make to make it more consistent (like create functions starting with as_ and is_ instead of as. is. As in base R). In general however inspiration for naming comes from data.table where you have a lot of functions starting with f-, standard eval versions of functions ending on -v, and you have capitalized abbreviations like setDT, .N, .SD, .SDcols etc.). So with the exception of some as. And is. Functions which mimic base R, I don’t think the namespace is that horrific. But perhaps if you can make concrete renaming suggestions, I can get inspired.

**Comment by ShixiangWang** [comment]
The data.table mainly use camel case:

```r
> library(data.table)
data.table 1.14.0 using 1 threads (see ?getDTthreads).  Latest news: r-datatable.com
> ls("package:data.table")
  [1] ":="                 "%between%"          "%chin%"             "%flike%"           
  [5] "%ilike%"            "%inrange%"          "%like%"             "address"           
  [9] "alloc.col"          "as.data.table"      "as.IDate"           "as.ITime"          
 [13] "as.xts.data.table"  "between"            "chgroup"            "chmatch"           
 [17] "chorder"            "CJ"                 "copy"               "cube"              
 [21] "data.table"         "dcast"              "dcast.data.table"   "fcase"             
 [25] "fcoalesce"          "fifelse"            "fintersect"         "first"             
 [29] "foverlaps"          "frank"              "frankv"             "fread"             
 [33] "frollapply"         "frollmean"          "frollsum"           "fsetdiff"          
 [37] "fsetequal"          "fsort"              "funion"             "fwrite"            
 [41] "getDTthreads"       "getNumericRounding" "groupingsets"       "haskey"            
 [45] "hour"               "IDateTime"          "indices"            "inrange"           
 [49] "is.data.table"      "isoweek"            "key"                "key<-"             
 [53] "last"               "like"               "mday"               "melt"              
 [57] "melt.data.table"    "merge.data.table"   "minute"             "month"             
 [61] "nafill"             "quarter"            "rbindlist"          "rleid"             
 [65] "rleidv"             "rollup"             "rowid"              "rowidv"            
 [69] "second"             "set"                "setalloccol"        "setattr"           
 [73] "setcolorder"        "setDF"              "setDT"              "setDTthreads"      
 [77] "setindex"           "setindexv"          "setkey"             "setkeyv"           
 [81] "setnafill"          "setnames"           "setNumericRounding" "setorder"          
 [85] "setorderv"          "shift"              "shouldPrint"        "SJ"                
 [89] "tables"             "test.data.table"    "timetaken"          "transpose"         
 [93] "truelength"         "tstrsplit"          "uniqueN"            "update.dev.pkg"    
 [97] "wday"               "week"               "yday"               "year"     
```

In general, this package create function names like above. But some functions break the rules.

![image](https://user-images.githubusercontent.com/25057508/117301547-52672a00-aead-11eb-8ae9-d8e87cf9551b.png)

And I also found some functions are hard to understand and remember.

![image](https://user-images.githubusercontent.com/25057508/117301794-94906b80-aead-11eb-9730-6218b8a5d53e.png)

My opinion is that names of the functions should be consistent both in word formation and semantics.

For example, `fgroup_by` to `fgroupBy`; `has_elem` to `hasElem`, `qsu` to `crossSummary`.

In general, all functions should stick to the format:

```
abc
ABC
aBc
is.a
is.ABC
is.aBc
```

I see the functions are categoried, so may you can directly set the function prefix by the short names of the categories to make it more easy to call.

I admit that this is not a easy work, but if you have spare time, I think it worth a try.

**Comment by SebKrantz** [comment]
Hi Shixiang, thanks a lot for your suggestion! Looking at the data.table namespace, I'd say most functions have a non-standard naming convention I also adopted with f-prefiv to prevent clashes, v-postfix for standard eval versions, and set* prefix to do things by reference. The core functions are lowercase word combinations. In my case I have mixed in some underslash functions essentially because of tidyverse compatibility end easy syntax comparison (e.g. fgroup_by). What I can do is get rid of all the dots as well as unexpected capitalizations by renaming

`as.factor_GRP` -> `as_factor_GRP`, `is.GRP` -> `is_GRP`, `is.qG` -> `is_qG`, `is.unlistable` -> `is_unlistable`, `is.categorical` -> `is_categorical`, `is.Date` -> `is_date`, `as.numeric_factor` -> `as_numeric_factor`, `as.character_factor` -> `as_character_factor`, 
`Date_vars` -> `date_vars`, 

I could also rename `fNobs` -> `fnobs`, `fNdistinct` -> `fndistinct`, `pwNobs` -> `pwnobs`, `fHDwithin` -> `fhdwithin`
`fHDbetween` -> `fhdbetween`, although I'm not sure if it's worth it as these functions are widely used, and I'd have to keep the original ones for a very long time (2 years +). 

Other function names I think will be kept. The quick-conversion functions `qDF` `qDT`, `qM`, `qF`, `qG` are consistent and in-line with *data.table* (`setDT` etc.), selector functions `get_vars`, `num_vars`, ..., `get_elem` etc. are consistent. and similarly the operators `L`, `F`, `D`, `Dlog`, `G`, `B`, `W`, `HDB`, `HDW`. I'll keep `GRP`, `BY` and `TRA`, for lack of better names, parsimony and because they are central to the package. The camel case will be kept in helper functions `setDimnames` etc. because they work like *stats* `setNames` and do not modify the argument by reference (like `settransform` or `setrename` and various *data.table* functions). The same applies to functions like `copyAttrib` and `copyMostAttrib` which are also exports of like-named functions in the C API and thus kept as they are. 

So in summary: What I can offer is to eleminate the "." and some unexpected capital letters in certain functions. The package will still have some all-uppercase functions (appliying to function abbreviations / operators) and some underslash functions, which came about by some essential design choices I have made and which I largely still support for reasons of parsimony.  

Let me know what you think. Others following this can comment as well.

**Comment by ShixiangWang** [comment]
Thanks for your response.

You are doing well. Indeed, it is not easy to follow all conventions. 

Please also try renaming functions like `qsu`, it is really hard to understand and remember.

**Comment by SebKrantz** [comment]
Hi, thanks for the feedback. So this will the probably be implemented in the next update. I'm sorry but I won't rename `qsu`, which stands for quick-summary, and is widely used as far as I am aware.

**Comment by ShixiangWang** [comment]
Ohh I can understand this when you say it is for quick summary. Maybe a alias qsummary is much easy to remember. However if you always explain how the short names came from in fun titles and cheatsheets，it would much better

Thanks for the explain ：）

**Comment by SebKrantz** [comment]
Sure, I refer you to the function documentation though, which explains this in most cases e.g. `?qsu`, `?varying` etc.
